testing/testing.go

```diff
@@ -113,22 +114,32 @@ func EnsureNodeSync(nodeManager common.NodeManager) {
 	defer timeout.Stop()
 	ticker := time.NewTicker(1 * time.Second)
 	defer ticker.Stop()
-
 	for {
 		select {
 		case <-timeout.C:
 			panic("timeout during node synchronization")
 		case <-ticker.C:
 			downloader := les.Downloader()
-
-			if downloader != nil {
-				isSyncing := downloader.Synchronising()
-				progress := downloader.Progress()
-
-				if !isSyncing && progress.HighestBlock > 0 && progress.CurrentBlock >= progress.HighestBlock {
-					return
-				}
+			if downloader == nil {
+				continue
+			}
+			if nodeManager.PeerCount() == 0 {
+				log.Debug("No establishished connections with a peers, continue waiting for a sync")
+				continue
 			}
+			if downloader.Synchronising() {
+				log.Debug("synchronization is in progress")
+				continue
+			}
+			progress := downloader.Progress()
+			if progress.CurrentBlock >= progress.HighestBlock {
```

Comment:
I would recommend running `statusd -les -http` and checking how the sync progress looks like using https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_syncing. As far as I remember, with LES it's a bit strange and that's why `progress.HighestBlock > 0` was added.

Maybe these checks are good, though but it would be great to confirm.

Reply:
It uses the same source of info as we are https://github.com/ethereum/go-ethereum/blob/master/internal/ethapi/api.go#L79-L94, so I just printed the progress:

```
INFO [01-23|16:38:58] Imported new block headers               count=384 elapsed=112.830ms number=1226112 hash=f197f6…495d1f ignored=0
INFO [01-23|16:38:58] Imported new block headers               count=192 elapsed=57.979ms  number=1226304 hash=ce5203…750e53 ignored=0
INFO [01-23|16:38:58] synchronization is not finished yet: current block 1226304 < highest block 1643240 geth=StatusIM
INFO [01-23|16:38:58] Imported new block headers               count=576 elapsed=165.479ms number=1226880 hash=60bd3c…9ba1e4 ignored=0
INFO [01-23|16:38:58] Imported new block headers               count=192 elapsed=55.092ms  number=1227072 hash=fcbc33…ef23af ignored=0
INFO [01-23|16:38:59] Imported new block headers               count=960 elapsed=283.560ms number=1228032 hash=cf13d1…68d1ba ignored=0
INFO [01-23|16:38:59] Imported new block headers               count=384 elapsed=116.356ms number=1228416 hash=db9630…6e83b3 ignored=0
INFO [01-23|16:38:59] Imported new block headers               count=768 elapsed=212.664ms number=1229184 hash=87f2da…b5ad22 ignored=0
INFO [01-23|16:38:59] synchronization is not finished yet: current block 1229184 < highest block 1643240 geth=StatusIM
INFO [01-23|16:38:59] Imported new block headers               count=768 elapsed=215.851ms number=1229952 hash=27dfbb…2ea2b3 ignored=0
```

Reply:
This looks good. Let's stay with this then.